setup.py

```diff
@@ -144,39 +138,53 @@ def package_check(*args, **kwargs):
             [['-msse2', '-mfpmath=sse'], [], simple_test_c, 'USING_GCC_SSE2'],
             [['-fopenmp'], ['-fopenmp'], omp_test_c, 'HAVE_OPENMP']], 'dipy')
 
-# Installer that checks for install-time dependencies
-class installer(install.install):
-    def run(self):
-        package_check('numpy', NUMPY_MIN_VERSION)
-        package_check('scipy', SCIPY_MIN_VERSION)
-        package_check('nibabel', NIBABEL_MIN_VERSION)
-        install.install.run(self)
+# Hard and soft dependency checking
+# pkg_import_name, version_str, setuptools_field, heavy_yes_no
+cython_dep = ((('Cython', info.CYTHON_MIN_VERSION, 'setup_requires', False),)
+              if need_cython else ())
+DEPS = (
+    (('numpy', info.NUMPY_MIN_VERSION, 'setup_requires', True),) +
+    cython_dep +
+    (('scipy', info.SCIPY_MIN_VERSION, 'install_requires', True),
+     ('nibabel', info.NIBABEL_MIN_VERSION, 'install_requires', False)))
+
+for name, min_ver, req_type, heavy in DEPS:
+    found_ver = get_pkg_version(name)
+    ver_err_msg = version_error_msg(name, found_ver, min_ver)
+    if not using_setuptools:
+        if ver_err_msg != None:
+            raise RuntimeError(ver_err_msg)
+    else:  # Using setuptools
+        # Add packages to given section of setup/install_requires
+        if ver_err_msg != None or not heavy:
```

Comment:
"Heavy" means that it's a pain to install from scratch.

In the code, it means that, if the package is installed already and has a
good-enough version, we don't add it to the setuptools requirements.

This avoids two undesirable cases:

* installing e..g. scipy with pip just because the current installation was
not installed with pip / setuptools;
* upgrading e..g. scipy when upgrading dipy (pip by default upgrades all
dependencies at the same time as the main package).